pkg/splunk/enterprise/configuration.go

```diff
@@ -556,20 +583,28 @@ func updateSplunkPodTemplateWithConfig(podTemplateSpec *corev1.PodTemplateSpec,
 		}
 	}
 
+	// Explicitly set the default value here so we can compare for changes correctly with current statefulset.
+	secretVolDefaultMode := int32(corev1.SecretVolumeSourceDefaultMode)
+
 	// add defaults secrets to all splunk containers
 	addSplunkVolumeToTemplate(podTemplateSpec, "secrets", corev1.VolumeSource{
 		Secret: &corev1.SecretVolumeSource{
-			SecretName: GetSplunkSecretsName(cr.GetIdentifier(), instanceType),
+			SecretName:  GetSplunkSecretsName(cr.GetIdentifier(), instanceType),
+			DefaultMode: &secretVolDefaultMode,
```

Comment:
Since the default mode is already set above (validateVolumes) can't it be retrieved from the CR spec here instead of initializing with corev1.SecretVolumeSourceDefaultMode.

Reply:
validateVolumes() validates/set what the user specified in the CR. The addSplunkVolumeToTemplate() adds **additional** volumes after what the user adds in the CR.